openstacksdk_plugin/decorators.py

```diff
@@ -57,15 +61,38 @@ def get_property_by_name(property_name):
 
             client_config = get_property_by_name('client_config')
             resource_config = get_property_by_name('resource_config')
-            resource_id = get_property_by_name(RESOURCE_ID)
-            if resource_id:
-                resource_config['id'] = resource_id
+            resource = class_decl(client_config=client_config,
+                                  resource_config=resource_config,
+                                  logger=ctx.logger)
+
+            # # Check if "use_external_resource" is set to True
+            is_external = get_property_by_name(USE_EXTERNAL_RESOURCE_PROPERTY)
+            if is_external and operation_name in ['create', 'delete']:
+                ctx.logger.info(
+                    'Using external resource {0}'.format(RESOURCE_ID))
+
+                # Get the remote resource
+                try:
+                    resource = resource.get()
+                    ctx.logger.info('This Object is {0}'.format(resource.name))
+                except exceptions.SDKException as error:
+                    _, _, tb = sys.exc_info()
+                    raise NonRecoverableError(
+                        'Failure while trying to request '
+                        'NetApp API: {}'.format(error.message),
+                        causes=[exception_to_error_cause(error, tb)])
+
+                # Check the operation type and based on that decide what to do
+                if operation_name == 'create':
+                    ctx.instance.runtime_properties[RESOURCE_ID] = resource.id
```

Comment:
Maybe add a logger message similar to that of "delete"?

Reply:
sure, will update it